### ISSUE TITLE
About the correspondence between the phases proposed in the paper and the phases in the code

### ISSUE THREAD

**Issue by luhexin** [yaoyao-liu/meta-transfer-learning#62]
I appreciate your contribution, and I've been studying this paper recently, but have a few questions

**Question 1：**
In the paper，pipeline of proposed few-shot learning method, including three phases: 
(a) DNN training on large-scale data 
(b) Meta-transfer learning
(c) meta-test
<img width="996" alt="framework" src="https://github.com/yaoyao-liu/meta-transfer-learning/assets/43848265/21b46ebf-a55a-4844-a2de-6bbcb594e8db">

In README.md you mention the pre-train phase
 meta-train phase and meta-test phase
![A%5M6RQUV)(C@4X27Q_}1FT](https://github.com/yaoyao-liu/meta-transfer-learning/assets/43848265/64c19dc8-1b8e-45d0-8915-26d7b8f74f7c)
Is the pre-training phase equivalent to DNN training on large-scale data? 
meta-train phase = Meta-transfer learning?
meta-test phase = meta-test?

**Question 2：**
https://github.com/yaoyao-liu/meta-transfer-learning/blob/835b6bbac9fb81ce2ce7de89cbe367aaf7bfd42c/pytorch/trainer/meta.py#L239-L294

Does the above code correspond to “classifier fine-tuning” in the (c)meta-test phase? 
Line 258 in this code sets the model to eval mode and does not fine-tune the base-learner

**Question 3：**
Do the train samples in (a) large-scale training and (b) meta-test both correspond to the same training set in the code?

**Comment by luhexin** [comment]
To add to Question 3, what I mean is that the data set in the code is divided into three parts: training set, validation set and test set. The train samples in the figure are sampled from the training set?

**Comment by yaoyao-liu** [comment]
Hello @luhexin 

Thanks for your interest in our work. The answers to your questions are as follows:

- **Answer to Q1**: Yes.
- **Answer to Q2**: The base learner is fine-tuned by the following line: https://github.com/yaoyao-liu/meta-transfer-learning/blob/835b6bbac9fb81ce2ce7de89cbe367aaf7bfd42c/pytorch/trainer/meta.py#L283
- **Answer to Q3**: For example, on miniImageNet, the model is trained on the **meta-train set**, including **pre-training** and **meta-transfer learning**. Then, the model is evaluated on the **meta-test set**. The validation set is never used. This is to align with other related work, e.g., MAML and ProtoNets. 

If you have any further questions, please feel free to contact me.

Best,

Yaoyao
